File: Practice/28/C++/28/28.cpp
#include <iostream>
using namespace std;

void print_factorization(unsigned int n) {
    int count = 0;
    if (n > 1) {
        for (int i = 2; i <= n; i++) {
            while (n % i == 0) {
                count++;
                n = n / i;
            }
            if (count > 0) {
                cout << i;
                if (count > 1) cout << "^" << count;
                if (n > 1) cout << "*";
            }
            count = 0;
        }
    }
    else if (n == 1) cout << "-";
}

int main()
{
    setlocale(LC_ALL, "russian");
    int n;
    cout << "Введите число для факторизации\n";
    cin >> n;
    print_factorization(n);
}

File: Practice/28/Python/28.py
def print_factorization(n):
    count = 0
    if n > 1:
        for i in range(2, n):
            while n % i == 0:
                count += 1
                n = n / i
            if count > 0:
                print(i, end='')
                if count > 1:
                    print("^", count, end="", sep="")
                if n > 1:
                    print("*", end="", sep="")
            count = 0
    elif n == 1:
        print("-")
n = int(input("Введите число для факторизации\n"))
print_factorization(n)
